integrationtest/backend/elasticsearch/src/test/java/org/hibernate/search/integrationtest/backend/elasticsearch/tmp/ExistsPredicateIT.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.elasticsearch.tmp;
+
+import static org.hibernate.search.util.impl.integrationtest.common.assertion.SearchHitsAssert.assertThat;
+import static org.hibernate.search.util.impl.integrationtest.common.stub.mapper.StubMapperUtils.referenceProvider;
+
+import java.util.List;
+
+import org.hibernate.search.engine.backend.common.DocumentReference;
+import org.hibernate.search.engine.backend.document.DocumentElement;
+import org.hibernate.search.engine.backend.document.IndexFieldReference;
+import org.hibernate.search.engine.backend.document.IndexObjectFieldReference;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaElement;
+import org.hibernate.search.engine.backend.document.model.dsl.IndexSchemaObjectField;
+import org.hibernate.search.engine.backend.document.model.dsl.ObjectFieldStorage;
+import org.hibernate.search.engine.backend.work.execution.spi.IndexIndexingPlan;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.rule.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.common.stub.mapper.StubMappingIndexManager;
+import org.hibernate.search.util.impl.integrationtest.common.stub.mapper.StubMappingScope;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class ExistsPredicateIT {
+
+	private static final String INDEX_NAME = "IndexName";
+
+	// this document is empty
+	private static final String DOCUMENT_0 = "0";
+
+	// this document has only first level fields: string and numeric
+	private static final String DOCUMENT_1 = "1";
+
+	// this document has also an empty nested and an empty flattened objects
+	private static final String DOCUMENT_2 = "2";
+
+	// this document has also second level fields:
+	// string and numeric within both nested and flattened objects
+	private static final String DOCUMENT_3 = "3";
```

Comment:
I think there's another interesting case that you forgot: a document with an object field whose only content is another object field, which contains a string field with a value.

E.g.:

```
{
   "nested": {
      "nested2": {
         "string": "foo"
      }
   }
}
```

What if we use the `exists` predicate on `nested` with this data? The document should match, but I'm not sure it would with the current implementation.

We should also test this with a flattened object field.

Reply:
I think you're right: current implementation does not. Let's handle the recursive case.